include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -0,0 +1,508 @@
+     *
+     *~~~~~~~~~~~~~~~{.cpp}
+     *     int main(int argc, char ** argv)
+     *     {
+     *          seqan3::argument_parser myparser(argc, argv); // initialize
+     * 
+     *          myparser.info.app_name = "Penguin_Parade";
+     *          myparser.info.version = "2.0.0";
+     *          myparser.info.short_description = "Organize your penguin parade";
+     *          myparser.info.description.push_back("First Paragraph.");
+     *          myparser.info.description.push_back("Second Paragraph.");
+     *          myparser.info.example.push_back("./penguin_parade -d 10 -m 02 -y 2017 --number_of_penguins 100");
+     * 
+     *          return 0;
+     *     }
+     *~~~~~~~~~~~~~~~
+     *
+     * \cond DEV
+     * The meta information are needed in all the format classes (parse_format,
+     * help_format, ...etc) and the argument_parser, so they can't be part of the 
+     * argument_parser class itself but need to be independent.
+     * \endcond
+     *
+     */
+    meta_information info;
+
+// #ifdef SEQAN_VERSION_CHECK_OPT_IN
+//     bool  enabled_version_check{false};
+// #else  // Make version update opt out.
+//     bool  enabled_version_check{true};
+// #endif  // SEQAN_VERSION_CHECK_OPT_IN
+    // std::future<bool> appVersionCheckFuture;
+
+private:
+    //!\privatesection
+
+    /*!\brief Initilizes the argument_parser class on construction
+     *
+     * \throws seqan3::parser_invalid_argument
+     *
+     * \details
+     * This function adds all command line parameters to a std::vector<std::string>
+     * to take advantage of the vector functionality later on. Additionally,
+     * the format member variable is set, depending on which parameters are given
+     * by the user:
+     *
+     * - **-h/--help** sets the format to seqan3::detail::help_format
+     * - **-hh/--advanced-help** sets the format to seqan3::detail::help_format
+     *                           and show_advanced_options to `true`.
+     * - **-export-help html** sets the format to seqan3::detail::html_format
+     * - **-export-help man** sets the format to seqan3::detail::man_format
+     * - **-export-help ctd** sets the format to seqan3::detail::ctd_format
+     * - else the format is that to seqan3::detail::parse_format
+     *
+     * If `-export-help` is specified with a value other than html/man or ctd
+     * a parser_invalid_argument is thrown.
+     */
+    void init(int argc, char ** argv)
+    {
+        for(int i = 1; i < argc; ++i) // start at 1 to skip binary name
+        {
+            std::string arg{argv[i]};
+            argv_.push_back(arg);
+
+            if (arg == "-h" || arg == "--help")
+            {
+                format = detail::help_format(false);
+                return;
+            }
+            else if (arg == "-hh" || arg == "--advanced-help")
+            {
+                format = detail::help_format(true);
+                return;
+            }
+            else if (arg == "--export-help")
```

Comment:
(I think) No, for the following reasons:

In the init() function, all (and only) format related arguments are checked. This is done on construction where NO further information apart from argc, argv is known.
In the parse() function, all special options are checked that require a "helpful" user message. For example the version and app_name information are not known on construction but when calling parse() they are.

The only thing that I could do is check for the arguments in the init() function and than set a bool `seen_version_arg` to true and then check the bool when calling parse()...